src/components/profile/profile.components.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from "react";
+import Data from "./friends";
+import Header from "./header/header.profile";
+import Friends from "./friends/friends.profile";
+import friends from "../chatroom/friends";
+
+function Profile() {
+  let [friendsList, setFriendsList] = useState(Data);
+  let [searchData, setSearchData] = useState("");
+  let tempFriendsList = friendsList.filter(
+    (friend) =>
+      friend.name.includes(searchData) ||
+      friend.description.includes(searchData)
+  );
+  function onSearchChange(e) {
+    setSearchData(e.target.value);
+  }
```

Comment:
```suggestion
  function onSearchChange(e) {
    setSearchData(e.target.value);
    setFriendList(friendList.fllter((friend) => {
      friend.name.includes(searchData) ||
      friend.description.includes(searchData)
    }))
  }
```

컴포넌트의 상태는 무조건 hook을 통해서만 관리해야 합니다! 리액트에 의해 관리되지 않는 상태는 의도치 않은 버그를 발생시킬 가능성이 있습니다.